plugins/rich-editor/src/scripts/Editor/Generic/PopoverController.tsx

```diff
@@ -5,73 +5,78 @@
  */
 
 import React from "react";
-import * as PropTypes from "prop-types";
-import uniqueId from "lodash/uniqueId";
 import { closeEditorFlyouts, CLOSE_FLYOUT_EVENT } from "../../Quill/utility";
 import { withEditor, IEditorContextProps } from "../ContextProvider";
+import { getRequiredID, uniqueIDFromPrefix, IRequiredComponentID } from "@core/Interfaces/componentIDs";
 
 export interface IPopoverControllerChildParameters {
+    id: string;
     initialFocusRef: React.RefObject<any>;
     blurHandler: React.FocusEventHandler<any>;
     isVisible: boolean;
     closeMenuHandler(event?: React.SyntheticEvent<any>);
 }
 
 interface IProps extends IEditorContextProps {
+    id: string;
+    contentID: string;
     classNameRoot: string;
     icon: JSX.Element;
     children: (props: IPopoverControllerChildParameters) => JSX.Element;
     onClose?: () => void;
 }
 
 interface IState {
+    id: string;
     isVisible: boolean;
+    contentID: string;
 }
 
 export default class PopoverController extends React.PureComponent<IProps, IState> {
-    private popoverContainerId: string;
-
     private initalFocusRef: React.RefObject<any>;
     private buttonRef: React.RefObject<HTMLButtonElement>;
     private controllerRef: React.RefObject<HTMLDivElement>;
 
     constructor(props) {
         super(props);
-
         this.controllerRef = React.createRef();
         this.initalFocusRef = React.createRef();
         this.buttonRef = React.createRef();
 
         this.state = {
+            id: getRequiredID(props, "popover"),
             isVisible: false,
+            contentID: props.contentID,
         };
+    }
 
-        this.popoverContainerId = uniqueId("richEditor-popover-");
+    get componentID() {
+        return this.state.id + "-contents";
     }
 
     public render() {
         return (
             <div className={this.props.classNameRoot} ref={this.controllerRef}>
                 <button
+                    id={this.state.id}
                     onClick={this.togglePopover}
                     onBlur={this.blurHandler}
                     className="richEditor-button"
                     type="button"
-                    aria-controls={this.popoverContainerId}
+                    aria-controls={this.componentID}
                     aria-expanded={this.state.isVisible}
                     aria-haspopup="true"
                     ref={this.buttonRef}
                 >
                     {this.props.icon}
                 </button>
-                <div id={this.popoverContainerId}>
```

Comment:
I don't understand your feedback here. We had the flyout contents wrapped inside a div with an id. Since we had a lot of our accessibility attributes on a child element, I thought it could cause problems. The button points to an element without all the features we want to offer. I got rid of the wrapper, and I just pass the ID to the child element.